src/s_fileops.h

```diff
@@ -0,0 +1,47 @@
+#ifndef S_FILEOPS_H
+#define S_FILEOPS_H
+
+/* Copyright (c) 2020 Andi McClure.
+* For information on usage and redistribution, and for a DISCLAIMER OF ALL
+* WARRANTIES, see the file, "LICENSE.txt," in this distribution.  */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdarg.h>
+
+typedef intptr_t t_fileops_handle;
```

Comment:
You can't simply change the `int` fd argument to `intptr_t` without breaking the ABI. Why not simply keep the `int` for all functions and treat it as an abstract index? For "real" file systems, it is the file descriptor, and for virtual file systems it would be an array index or hashmap key.